SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+    #####            print ZZ,  qq, ii
+    #####            print  g_dbase['ff'][ ii , : ]
+    ##            ii = ii + 1
+    ##    g_dbase['Sq_halfQ'] = numpy.array( [ 0 , 0 , 0 ] ) ;
+    ##    g_dbase['Sq_bound'] = numpy.array( [ 0 , 0 , 0 ] ) ;
+    ##    g_dbase['Sq_free']  = numpy.array( [ 0 , 0 , 0 ] ) ;
+    ###    g_dbase['ph_sigma'] ;
+
+    def f_load_snp_content(self, a_fp, a_snp):
+        dbase_root = (
+            "/data/snp_" + str(a_snp).zfill(self.g_s2e["setup"]["num_digits"]) + "/"
+        )
+        xsnp = dict()
+        xsnp["Z"] = a_fp.get(dbase_root + "Z")[()]
+        xsnp["T"] = a_fp.get(dbase_root + "T")[()]
+        xsnp["ff"] = a_fp.get(dbase_root + "ff")[()]
+        xsnp["xyz"] = a_fp.get(dbase_root + "xyz")[()]
+        xsnp["r"] = a_fp.get(dbase_root + "r")[()]
+        N = xsnp["Z"].size
+        xsnp["q"] = numpy.array(
+            [xsnp["ff"][numpy.nonzero(xsnp["T"] == x)[0], 0] for x in xsnp["xyz"]]
+        ).reshape(
+            N,
+        )
+        xsnp["snp"] = a_snp
+
+        return xsnp
+
+    def f_load_snp_xxx(self, a_real, a_snp):
+        xfp = h5py.File(
+            self.g_s2e["prj"]
+            + "/pmi/pmi_out_"
+            + str(a_real).zfill(self.g_s2e["setup"]["num_digits"])
+            + ".h5",
+            "r",
+        )
+        xsnp = self.f_load_snp_content(xfp, a_snp)
+        xfp.close()
+        return xsnp
+
+    def f_load_snp_from_dir(self, path_to_snapshot):
+        """Load xmdyn output from an xmdyn directory.
+
+        :param path: The directory path to xmdyn output.
+        :type path: str
+
+        :param snapshot_index: Which snapshot to load.
+        :type snapshot_index: int
+
+        :returns: The snapshot data.
+        :rtype: dict
+
+        """
+
+        xsnp = dict()
+        xsnp["Z"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "Z.dat")
+        )  # Atom numbers
+        xsnp["T"] = numpy.loadtxt(os.path.join(path_to_snapshot, "T.dat"))  # Atom type
+        xsnp["uid"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "uid.dat")
+        )  # Unique atom ID.
+        xsnp["r"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "r.dat")
+        )  # Cartesian coordinates.
+        xsnp["v"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "v.dat")
+        )  # Cartesian velocities.
+        xsnp["m"] = numpy.loadtxt(os.path.join(path_to_snapshot, "m.dat"))  # Masses.
+        xsnp["q"] = numpy.loadtxt(os.path.join(path_to_snapshot, "q.dat"))  # Ion charge
+        xsnp["ff"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "f0.dat")
+        )  # Form factors of each atom type.
+        xsnp["Q"] = numpy.loadtxt(
+            os.path.join(path_to_snapshot, "Q.dat")
+        )  # Wavenumber grid for form factors.
+
+        return xsnp
+
+    def f_load_sample(self, sample_path):
+        sample = dict()
+        xfp = h5py.File(sample_path, "r")
+        xxx = xfp.get("Z")
```

Comment:
Variable names are somewhat confusing to me, but that may be only because I'm not familiar with PMI.

Reply:
This is a heritage, just leave it like this at this moment.